packages/client/src/containers/LandingPage/useFilteredTools.js

```diff
@@ -0,0 +1,99 @@
+import { useState, useCallback, useEffect, useMemo } from 'react';
+import getApiBaseUrl from '../../utils/getApiBaseURL';
+
+export function useFilteredTools() {
+  const [tools, setTools] = useState([]);
+  const [baseUrl, setBaseUrl] = useState(`${getApiBaseUrl()}/api/tools`);
+  const [isLoading, setIsLoading] = useState(true);
+
+  // fetching all tools
+  const getAllTools = useCallback(() => {
+    function fetchAllTools() {
+      setIsLoading(true);
+      setBaseUrl(`${getApiBaseUrl()}/api/tools`);
+      const promise = fetch(baseUrl).then((response) => response.json());
+      return promise;
+    }
+    fetchAllTools().then((data) => {
+      setIsLoading(false);
+      setTools(data);
+    });
+  }, [baseUrl]);
+
+  // Modifying a query string, by adding a new query param. Triggers by pressing on a checkbox
+  const addQueryParam = useCallback(
+    (e, fetchKey) => {
+      const url = `${fetchKey}[]=${e.target.value}`;
+      // Check if it gonna be a first query param
+      const isIncludes = baseUrl.includes('?');
+      if (!isIncludes) {
+        setBaseUrl((prevUrl) => prevUrl.concat(`?${url}`));
+      } else {
+        setBaseUrl((prevUrl) => prevUrl.concat(`&${url}`));
+      }
+    },
+    [baseUrl],
+  );
+
+  // Modifying a query string, by deleting an already existing query param. Triggers by pressing on the same checkbox on a second time
+  const removeQueryParam = useCallback(
+    (e, fetchKey) => {
+      const url = `${fetchKey}[]=${e.target.value}`;
+
+      // All possible query params
+      const queryParams = {
+        queryParamBeginWithQuestionMark: `?${url}`,
+        queryParamBeginWithAmpersand: `&${url}`,
+        queryParamEndWithAmpersand: `${url}&`,
+      };
+
+      function checkIfBaseUrlAlreadyIncludeQueryParamToDelete(queries) {
+        for (const key in queryParams) {
+          if (Object.hasOwnProperty.call(queryParams, key)) {
+            const element = queryParams[key];
+            const isInclude = baseUrl.includes(element);
+            if (isInclude) {
+              const newBaseUrl = baseUrl.replace(element, '');
+              setBaseUrl(newBaseUrl);
+            }
+          }
+        }
+      }
+
+      checkIfBaseUrlAlreadyIncludeQueryParamToDelete(queryParams);
+    },
+    [baseUrl],
+  );
```

Comment:
Looking at the logic here, I feel like maybe you should reach for a higher level solution than string manipulation. I suggest we try to use https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams here instead. So this code might change to something like this:

```
const [filters, setFilters] = useState(new URLSearchParams());

const addQueryParam = useCallback((filter, key) => {
    // Make a copy to get a new reference 
    const newFilters = new URLSearchParams(filters);

    newFilters.append(key, filter);

    setFilters(newFilters);
}, []);

const removeQueryParam = useCallback((filter, key) => {
  const newFilters = new URLSearchParams(filters);

  // Grab the current values (if any), then delete the whole group
  const values = newFilters.getAll(key);
  newFilters.delete(key);

  // Filter out the selected value, and if there are any values left, add that as the new value for the group 
  const remainingValues = values.filter((value) => value !== filter);
  if (remainingValues.length) {
    newFilters.set(key, remainingValues);
  }

  setFilters(newFilters);  
}, []);

const getAllTools = useCallback(() => {
    setFilters(new URLSearchParams());
}, []);

useEffect(() => {
  async function fetchTools() {
      setIsLoading(true);
      // If the filters are empty, you'll just get a dangling ? at the end which is not a problem
      const response = await fetch(`${getApiBaseUrl()}/api/tools?${filters.toString()}`);
      const fetchedTools = await response.json();
      setTools(fetchedTools);
      setIsLoading(false);
    }
    
    fetchTools();
}, [filters]);
```